packages/calypso-e2e/src/lib/pages/media-page.ts

```diff
@@ -24,6 +26,10 @@ const selectors = {
 		`.image-editor__toolbar-button span:text("${ text }")`,
 	imageEditorResetButton: 'button[data-e2e-button="reset"]',
 	imageEditorCancelButton: 'button[data-e2e-button="cancel"]',
+
+	// When embedded in a modal
+	modalActionButton: ( action: ModalAction ) =>
+		`button[data-e2e-button="${ action.toLowerCase() }"]`,
```

Comment:
From my personal stylistic perspective, it feels a little bit off to have the modal handling in this media POM class. Maybe it violates the single responsibility principle? The modal dialog seems to me like something distinct from its contents. 

Two alternative suggestions:
- We might be able to just make a generic modal dialog component, as I believe we re-use the same modal DOM structure across Calypso.
- Alternatively, we could maybe make a class called MediaModal that has a property that gives access to the underlying media POM class contained within the modal?

What do you think?

Reply:
Looks like a comment I had left in this section of the code explaining my decision went missing. I'll summarize what I had written in it.

The gist was that I had decided to place the modal related actions within `MediaPage` for a couple of reasons:
- there are few, if any other places in Calypso that embed an entire Calypso page within a modal.
- I considered the modal to be just one another form in which the MediaPage can manifest itself.

My decision was driven primarily by the first point. While Calypso does use react modals throughout, we have so far created components for modals that are 'tailor made' for the component in question (eg. `PreviewComponent`, `RevisionComponent`).

With that said, if we are to change the implementation to a dialog modal component, then my vote would be on using dependency injection.

`const reactModalComponent = new ReactModalComponent( page, new MarketingPage(page) )`

then interact with the injected page object via:

`await reactModalComponent.pageObject.<method_name>`.

What do you think about these?

Reply:
Yeah, that reasoning makes a lot of sense! I like the style of the dependency injection you l aid out -- what's cool too is we could use generic typing on the argument of the POM class passed in to have type safety on the methods on the `pageObject`.

I think we'll eventually need the ReactModalComponent, but we can definitely punt that until later and we need it more (and not do it in this PR). I'll defer to you! 🙂

Reply:
Addressed in 0b1be87bc91b305ae5ac6f91c7b4a865431d7c49.